email/fetcher.go

```diff
@@ -313,9 +313,14 @@ func (f *Fetcher) persistMessage(mailbox *imap.MailboxStatus, msg *imap.Message,
 		return errors.AddContext(err, "could not read msg body")
 	}
 
-	// parse the 'from'
+	// parse the 'from', we evaluate the reply-to first because 'abuse@' emails
+	// are forwarded to our mailbox and the original sender is in the reply-to
+	// field, while the from field holds 'abuse@', so we lose the original
+	// context
 	from := "unknown"
-	if len(msg.Envelope.From) > 0 {
+	if len(msg.Envelope.ReplyTo) > 0 {
```

Comment:
I wonder if this can be used against us. e.g. somebody sending an email, trying to make us think someone else reported the abuse.

Reply:
definitely, emails in general are very weak in that sense, you can essentially alter just about any field in the email to try and mislead us. Currently these fields aren't used anywhere besides just from having some context in our database. I don't really think we should overthink this for the time being as we don't even use this context at the moment.

Also I wouldn't even know how to combat this really, like I said it's super easy to construct an email that looks like it's coming from an email address you don't own.

Reply:
Agreed. I'm not worried about changing the sender because there are all kinds of things preventing spoofing but tbh I don't know if they also protect the "ReplyTo" field.